tests/test_support.py

```diff
@@ -186,6 +188,31 @@ def test_parameters(self):
         # Should not raise any error, as repeat is an endless source of values
         x1, x2 = parameters('x1, x2', value=[2.0, 1.3], min=repeat(0.0))
 
+    def test_cached_property(self):
+        class A(object):
+            @cached_property
+            def f(self):
+                return 2
+
+        a = A()
+        # Deleta before a cache was set will fail silently.
+        del a.f
+        with self.assertRaises(AttributeError):
+            # Cache does not exist before f is called
+            a._f
+        self.assertEqual(a.f, 2)
+        self.assertTrue(hasattr(a, '_f'))
+        del a.f
+        # check that deletion was successful
+        with self.assertRaises(AttributeError):
+            # Does not exist before f is called
+            a._f
+        # However, the function should still be there
+        self.assertEqual(a.f, 2)
+        with self.assertRaises(AttributeError):
+            # Setting is not allowed.
+            a.f = 3
```

Comment:
Add a test in which A.f increments a counter so we can make sure the fucntion is not called unnecessarily.

Reply:
Good one

Reply:
Present below this conversation